unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs

```diff
@@ -365,6 +387,39 @@ public void AddHelpAndSupportWindow(HelpAndSupportHUDController controller)
         };
     }
 
+    public void AddGoToGenesisWindow(GoToGenesisPlazaHUDController controller)
+    {
+        if (controller == null || controller.view == null)
+        {
+            Debug.LogWarning("AddGoToGenesisWindow >>> Go to Genesis window doesn't exist yet!");
+            return;
+        }
+
+        goToGenesisHud = controller;
+        goToGenesisHud.view.OnClose += () =>
```

Comment:
There's no risk of dangling events here?

Reply:
All these subscriptions belong to HUDs directly related to the taskbar (which is will be always alive during the runtime), so in this case there is not exist the risk of dangling events. In fact we are doing the same with the rest of HUDs related to the taskbar (chat, friends, settings, etc.).